File: im/프로그래머스 Lv2/피보나치 수.java
class Solution {
    public int solution(int n) {


        if( n == 0 ) return 0;
        if( n == 1 ) return 1;

        long f1 = 0;
        long f2 = 1;
        long f3 = f2 + f1;

        for(int i = 0; i < n-2; i++){

            f1 = f2;
            f2 = f3;
            f3 = f1 + f2;
            f3 %= 1234567;
        }


        return (int)(f3 % 1234567);
    }
}

File: baik/Algorithm/lv2/피보나치 수.swift
//func solution(_ n:Int) -> Int {
//    return fiv(n) % 1234567
//}
//
//func fiv(_ n:Int) -> Int {
//    if n < 2 {
//        return n
//    }
//    else {
//        return fiv(n-1) + fiv(n-2)
//    }
//}

// 위 방식으로 재귀로 풀 경우 시간초과 발생

func solution(_ n:Int) -> Int {
    var fibonacci: [Int] = [1, 2]
    var i = 0
    while fibonacci.count < n {
        fibonacci.append((fibonacci[i] + fibonacci[i+1]) % 1234567)
        i += 1
    }
  
    return fibonacci[n-1]
}


// 멀리뛰기 알고리즘을 짜던중 n에 따라 피보나치로 증가함을 확인함
